### DISCUSSION TITLE
DiscriminatorMap integer key

### DISCUSSION THREAD

**Discussion by Hricer** [doctrine/doctrine2#9921] [General]
Why can't I use **integer** type key in DiscriminatorMap?

```php
#[ORM\Entity]
#[ORM\InheritanceType('SINGLE_TABLE')]
#[ORM\DiscriminatorColumn(name: 'discriminator', type: 'integer')]
#[ORM\DiscriminatorMap([
    1 => RequireReaction::class,
    2 => ResolveReaction::class])
]
abstract class AbstractReaction
{
}
```

>  Parameter #1 $value of attribute class Doctrine\ORM\Mapping\DiscriminatorMap constructor expects array<string, class-string>, array<int, string> given.

**Comment by greg0ire** [comment]
That restriction was added in https://github.com/doctrine/orm/pull/9274
If it works, consider sending a PR to relax this constraint. cc @derrabus

**Reply by derrabus** [reply]
Indeed, that annotation is too restrictive.

**Reply by derrabus** [reply]
#9922
